### ISSUE TITLE
Migrate Fragments in ViewPager to Jetpack Compose

### ISSUE THREAD

**Issue by angrezichatterbox** [scribe-org/Scribe-Android#150]
### Terms

- [X] I have searched [open and closed feature requests](https://github.com/scribe-org/Scribe-Android/issues?q=is%3Aissue+label%3Afeature)
- [X] I agree to follow Scribe-Android's [Code of Conduct](https://github.com/scribe-org/Scribe-Android/blob/main/.github/CODE_OF_CONDUCT.md)

### Description

We need to migrate the existing Fragments in the ViewPager2 (SettingsFragment, AboutFragment, and MainFragment) to Jetpack Compose. Additionally, the navigation bar and ViewPager2 host must be updated to align with the new Compose implementation. This migration will enhance performance and maintainability by utilizing the declarative UI approach of Compose.

More details of the migration can be found at the following link:

[CodeLabs for Jetpack Compose from XML](https://developer.android.com/codelabs/jetpack-compose-migration#0)

[Migration Strategy](https://developer.android.com/develop/ui/compose/migrate/strategy)

### Contribution

I would love to work on this and am happy to assist anyone fix this issue.

**Comment by realjoni17** [comment]
hi i want to contribute to this

**Comment by angrezichatterbox** [comment]
Yes, feel free to work on this issue. In case of queries feel free to ask them here :)  Thanks

**Comment by andrewtavis** [comment]
Looking forward to the contribution, @realjoni17!

**Comment by angrezichatterbox** [comment]
Hi @realjoni17, just checking in—are you still working on this issue? If you need any help, please don’t hesitate to reach out. We’re here to support you!

**Comment by andrewtavis** [comment]
Hey @realjoni17 👋 We would like to get to this issue soon, so hope it's ok that @angrezichatterbox takes over :) Please let us know if you'd like to work on another issue!

**Comment by andrewtavis** [comment]
CC @Jag-Marcel for collaboration with @angrezichatterbox on this issue :)

**Comment by angrezichatterbox** [comment]
I will explore the two key aspects discussed during the sync: the ability of ViewPager to handle both Compose and XML simultaneously. If feasible, we can create individual pages as separate issues. I will write down further steps here after I find more about that.

**Comment by angrezichatterbox** [comment]
We can definitely make it work with both fragments and Compose in ViewPager2. I’ll start by creating the base PR to update the main activity, set the content view to Compose, and get the gesture controls working between the two (Fragment and Compose).

Then, @Jag-Marcel, we can split the work on the Settings and About pages. If you want, feel free to grab one or both—I’m happy to split the work. 

Once the pages are migrated, we can look into switching from ViewPager2 to a Compose-native pager (like HorizontalPager) for a cleaner, more Compose-friendly approach.

Should I make a separate issue for these? or it could be done one by one considering the same issue? @andrewtavis

**Comment by andrewtavis** [comment]
I think separate issues makes sense, @angrezichatterbox! Please ping @Jag-Marcel in the one you'd envision for him :) Maybe once those issues are made we can close this one.

**Comment by angrezichatterbox** [comment]
This issue can be resolved once we complete #256.

**Comment by angrezichatterbox** [comment]
Closed by #281

**Comment by andrewtavis** [comment]
Thanks for the amazing work and the persistence to bring it all in, @Saifuddin53! Really some amazing work 😊

**Comment by Saifuddin53** [comment]
My pleasure @andrewtavis ☺️
